### ISSUE TITLE
Sprint - Oct 14 to Oct 25, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#25467]
# Global Sprint Planning

## 3 things that might take us down

1. Unpredictable ingestion lag
  2. Clickhouse

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Customer Driven People
Support: 

## Retro

- 🟡  Communicate action and destination upgrades @mariusandra 
  - We did a lot of communication
  - Stuck behind new pricing before we can ask _everyone_ to move over
- 🟢 Bring destinations 3000 out of feature preview @benjackwhite 
  - It's out
- ⚪ Documenting and updating runbooks for realtime destinations @benjackwhite 
  - ??
- 🔴 Pricing? @annaszell 
  - Focus this sprint

## OKRs
* Widespread usage of the new CDP 
* Messaging V1 
* Ingestion is rock solid

### High priority

- CDP pricing
- Messaging v0.1
- ???
-

**Comment by daibhin** [comment]
# Team Error Tracking

Time off:
@oliverb123 @neilkakkar maybe off 1-2 days
@daibhin @neilkakkar EU on-call week 1 each

## Retro

✅ Symbolify a single stacktrace @oliverb123 
 - New Rust service created and reading from Plugin Server
 - Local environments configured and demangled our first stack frame!
 - End-to-end test to run through entire service
 - Source map upload endpoint & S3 bucket ready to go

✅ Better filtering @neilkakkar
- Universal search added to exception query
- Added super property support to Python to support exception filtering by repository
- Searching by url, file, or task name

🟢 Start planning issues overrides table schema @daibhin 
- RFC in-progress for the approach we plan on taking to handle issue merging

✅ Enable autocapture on billing repo @daibhin
✅ Reached out to Python SDK customers. 2 people going through setup @daibhin
🟡 Rename stack trace properties on JS SDK (beware: Sentry integration relies on some of this) @neilkakkar 

## OKR

🟢 Human readable stack traces for Javascript
🟢 Ensure scalability of query time issue grouping
⚪️ Alerting via CDP
🟢 Launch

### High priority

Storing resolved source maps @oliverb123 
- Create new PostHog model for source map pointers (symbol set reference -> S3 URL) @neilkakkar 
- Writing / Reading from S3
- Storing / Invalidating symbol set results

Overrides table for issue merging @daibhin 
- Close out open questions on issue merging support in alerting
- Create ClickHouse table
- [Stretch] Adjust exceptions query to use new issues field rather than fingerprint

_Advanced_ fingerprinting in Rust @neilkakkar 
- Replicate fingerprinting logic from plugin server
- Use resolved frames for better fingerprinting
- Handle unresolved frame
- Add boolean to track resolution of a frame

### Low priority / side quests

- Add language tag to frames on the JS & Python client @neilkakkar @daibhin

- Storing frame resolution results
  - Key / Value store plan (do we go with Postgres or Dynamo or something else)

**Comment by Phanatic** [comment]
# Team Surveys

## Retro
- Customer support : 
  * Overall really good.
  * Met a lot of SLAs

✅  Ship & Beta testing no-code experiments with customers - @Phanatic 
✅  Fixed some issues with surveys and cohort filters from support hero coverage - @dmarticus 
✅ Various small UI fixes for Surveys and Flags - @jurajmajerik 

### Side Quests 
* Show a banner on Surveys page with `posthog-js` version in use not supporting survey features.

---

## OKRs

1. Implement Enterprise-ready features for Surveys
2. Increase adoption of Surveys.

### High priority
*  #17512
* Investigate support for mobile surveys in `posthog-react-native` library

**Comment by fuziontech** [comment]
# Team Where Do We Put Our Data? 
![image](https://github.com/user-attachments/assets/9b9485f0-2537-481a-9633-1ad17851b64b)

## OKR Q2 2024

### Objective

**Goals for Q4:**

- Improve elasticity and flexibility of our data store by putting all our data in Iceberg
    - [Work with Altinity](https://docs.google.com/document/d/12icx71-_kV0aICP3igtOc0di_yrey_W82HzbadtNPBY/edit?usp=sharing) effectively to ship read path for Iceberg on ClickHouse - Brett Hoerner
    - Setup infrastructure to ship all of our data to Iceberg on S3 - James Greenhill
    - Shipping Query logs to Iceberg - James Greenhill
- Continue improving CH operational expertise
    - Upgrade to a later version of Clickhouse - James Greenhill
    - Capacity planning - James Greenhill
    - Automation - Daniel Escribano
    - Put some of the basic mitigation operations in runbooks - Daniel Escribano
- Schema management
    - Tool for schema migration (coordinator schemas) - Daniel Escribano
    - Tool for long running mutations - Daniel Escribano
- Continued investment in performance
    - Tooling for other teams to understand which queries are slow and why - Ted Kaemming
    - Investigate variability of queries - Ted Kaemming
    - Per-team limits on queries/query complexity (needs product work) - Ted Kaemming

## Board
https://github.com/orgs/PostHog/projects/85/views/2

## Retro

### Outages
- EU Batch Exports using large amounts of memory + earlyoom chaos monkey
- CH node OOM due to UDF on US causing widespread issues in the cluster

- Write path for Iceberg setup to iterate ✅ 
- Backup cluster is moved to i4i.metal (more disk more io) ✅ 
- Upgraded to ClickHouse 24.4 ⚠️ 
  - Progress has been made, but the new query parser is much more strict with our queries
- We have a plan with Altinity ✅ 

@tkaemming - 
@Daesgar - 
@fuziontech - 
@bretthoerner - 

## Board Snapshot 

TL;DR

- Reformat 🧊 berge schema to accommodate Batch Export work
- Have a plan for Materialized Columns on 🧊 
- Setup TTL for events for 48 months to gp3 disk
- Complete upgrade to CH 24.4
- First PRs with Altinity

<img width="1081" alt="image" src="https://github.com/user-attachments/assets/f2fa5035-36ea-42d2-b51c-c833f0d4e73c">

**Comment by zlwaterfield** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

- @raquelmsmith
    - [ ] Implement Personless default switch
    - [ ] Keep working with new onboarding templates flow
    - [x]  Move activation events to the backend
    - [x] Help Patricio get up and running
    - [x] Work with data science contractor
- @zlwaterfield
	- [x] next round of emails from mailgun to c.io 
	- annual credits
		- [x] release to more users
		- [x] make the banner closable but still show on the page
		- [x] https://posthog.slack.com/archives/C01MGUHFH6G/p1726790198011829 - credits ui for annual customer and existing customers with a discount
	- RBAC
		- [x] Finish reviewing Ben's existing PR and RFC 
		- [ ] Finish v1 of the RFC for billing	
		- [ ] Start working on it - decide if we should continue on Ben's PR or start from scratch and do it piece by piece - get at least 1 PR merged for it
	- [ ] (from previous) ICP analysis - review completed work and improve charts/dashboard for new data flowing in from SalesForce
- @patricio-posthog 
	- [ ] 
</details>  

## Q4 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Get to a 4/10 in Enterprise platform support (Zach)
	- 🟡 Release Enterprise addon and iterate with sales to find a pricing model that makes sense
	- 🟡 Build RBAC
2. 🟡 Make revenue & forecasting more reliable (Patricio)
	- 🟡 Rejig how we do sync_invoices so we don’t see big drops - respond to webhooks instead of large syncs
	- ⚪ Evaluate temporal vs celery for usage reports and billing tasks so we can adhere to rate limits better
	- ⚪ Side hustle, if time permits: Rejig how anonymous / identified events are billed, as the current implementation is confusing.
3. 🟡 Activation: reverse trials, analytics onboarding flow, understand CC activation (Raquel)
	- ⚪ Dig into the data, watch replays to figure out why CC activation rate has been falling
	- ⚪ Add reverse trial to onboarding
	- 🟡 Add reverse trial to billing page for teams

## This sprint

Time off: Zach 1 day, Raquel 2 days

- @raquelmsmith 
	- [ ] Fix forecasting bug that says peoples' forecast is much higher than what they see on billing page
	- [ ] Fix bugs with product analytics onboarding flow
	- [ ] Personless default change
	- [ ] Make sure metrics move didn't break wnything
	- [ ] Re-up activation metrics for product analytics - try using `insight saved` instead of `insight created`
	- [ ] Use new product intent table in activation metrics
- @patricio-posthog 
	- [ ] Continue working with the sync invoices changes to use webhooks instead of a daily sync using Stripe API
	- [ ] Check billing limit bug raised by jdoodle (support ticket)
	- [ ] May investigate slow load times in billing page and try to optimize them (confirm first if it only affects ElevenLabs)
- @zlwaterfield 
	- RBAC
		- [ ]  Finish v1 of the RFC for billing
		- [ ]  Start working on it - decide if we should continue on Ben's PR or start from scratch and do it piece by piece - get at least 1 PR merged for it
	- [x] Finish up the Technical Exercise RFC - resolve comments and get merged, share out
	- [ ] Get the CC entry PR shipped
	- [ ] Adding user id to billing token for events
	- [x] Adding Hogfetti component
	- Growth review (if time permits)
		- [ ] We had a number of “accidental” signups that got refunded in sept, which might show up as churn
		- [ ] Teams plan survey after they subscribe
		- [ ] Use new ICP data to compare with CC activations - are high ICP people still subscribing at the same rate?
		- [x] Add is subscribed bool and mrr to org group

**Comment by danielxnj** [comment]
# Team Infra

## Retro / hangover

- ArgoCD all the way @danielxnj @bciaraldi 
  - 🟢  Argocd Rolling syncs
  - 🟢  Argocd diff workflow
  - 🟡 We want all internal k8s stuff deployed with ArgoCD (12 internal helm charts were migrated, 25% done)
  - 🟡  We want a PostHog service deployed with ArgoCD (Billing was migrated for dev)
- 🔴 PoC'ing out what a great secret management workflow would look like @bciaraldi 
 
## OKR

1. 🏃 Continuous Delivery
2. 💰 Cost visualization
3. 🧪 Continuous Integration

### High Priority
- ArgoCD all the way @danielxnj @bciaraldi 
  - Finish moving all internal helm charts to ArgoCD
  - Fully deploy a PostHog service deployed with ArgoCD 

### Side quest
- Drift detection

**Comment by Twixes** [comment]
# Team AI Product Assistant

**Support hero:** final support for Product Analytics @skoob13, then @Twixes 

## Retro

- ✅ New queries for retrieving relevant events/properties: event taxonomy query and property values query @skoob13 
- ✅ Productizing the ReAct approach to generation @skoob13
- ✅ Showing a set of suggested questions in the UI @Twixes

## Objective

LLM-based product manager as a service

### High priority

- Router between AI agent's skills (trends, funnels, SQL, etc.) @skoob13 
- Support for all insight types @skoob13 @Twixes 
- Basic SQL support @Twixes 
- Feedback button in answers (good answer/bad answer) @Twixes 

### Low priority / side quests

- Environments out as a feature preview @Twixes

**Comment by aspicer** [comment]
# Team Product Analytics

**Support hero:** final support for Product Analytics @skoob13, then @Twixes 

## Retro
- ✅ Both members have one week of support
- ✅ Anirudh to roll out alerts and make improvements based on user feedback
    - Well received, no major issues.
    -   From early user feedback, adding relative alerting and making changes to teh dynamics of the snoozing mechanism
- ✅ Sandy to get UDFs rolled out and work on cohort migration in free time
    - UDFs deployed and rolled out to team.
    - OOM issues discovered and resolved
    - Code framework for cohort migration built, will be increasing scope by adding property group types in one by one


## Sprint

Anirudh
- Create snoozing mechanic for firing alerts
- Launch relative alerts
- Roll out to more users (talk to comms about finding beta test customers)

Sandy
- UDF rollout for steps + moving funnel trends to UDF
- Get a subset of cohorts calculating on HogQL

**Comment by EDsCODE** [comment]
# Team Data House

TBD at offsite!

**Comment by dmarticus** [comment]
# Team Flags

## Retro

- Offsite
- Team split
- Support heroics
    - One bad CSAT (see context [here](https://posthog.slack.com/archives/C075D3C5HST/p1728148788234539), TL;DR customer was grouchy about a perceived outage when we didn't have one, didn't take kindly to my explanation.
- SLAs are all good

✅  Added metrics to the new feature flags service to track how well it's working in production without hitting DB, redis, etc
✅  Code references hackathon/deep dive on feasibility
    - Writeup is [here](https://posthog.slack.com/archives/C07Q2U4BH4L/p1728426991807349), TL;DR this is valuable but will require heavier lifting than simply using Github's search API.  Maybe be worth taking on as a stretch OKR once Haven joins and ramps up.

---

## OKRs

1. Ship new feature flags in production and accomplish the 10x scale + 2x speed boost performance targets (ideally decreases infra spend by N% as well, will try to figure out a more specific target going forward)

### High priority

- Finish cohort matching for new feature flag service
    - dynamic matching at the service level
    - static matching at the DB level


### Low Priority

- Finish copying over FF integration tests from the production repo (lots of valuable alpha in our existing `/decide` tests, but this doesn't need to happen this sprint)
